pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -206,25 +211,36 @@ func (w *Watcher) traversePluginDir(dir string) error {
 			if w.containsBlacklistedDir(path) {
 				return filepath.SkipDir
 			}
-
```

Comment:
starting watching the dir after we descend and scan it for socket files means that we can miss new socket files that appear. we *have* to start watching before we scan the content.

Reply:
Yes, there'd be another race if the file were created after we descend and before start watching.

I changed the order of events to:

1. Register file system watchers as we descend
2. Enqueue and process socket files found
3. Start processing socket files observed by watchers added in step 1

Reply:
Done.